crates/wallet/src/wallet/mod.rs

```diff
@@ -2283,44 +2280,18 @@ impl Wallet {
         Ok(())
     }
 
-    /// Commits all currently [`staged`](Wallet::staged) changes to the `persist_backend`.
-    ///
-    /// This returns whether anything was persisted.
-    ///
-    /// # Error
-    ///
-    /// Returns a backend-defined error if this fails.
-    pub fn commit_to<B>(&mut self, persist_backend: &mut B) -> Result<bool, B::WriteError>
-    where
-        B: PersistBackend<ChangeSet>,
-    {
-        let committed = StageExt::commit_to(&mut self.stage, persist_backend)?;
-        Ok(committed.is_some())
-    }
-
-    /// Commits all currently [`staged`](Wallet::staged) changes to the async `persist_backend`.
-    ///
-    /// This returns whether anything was persisted.
-    ///
-    /// # Error
-    ///
-    /// Returns a backend-defined error if this fails.
-    #[cfg(feature = "async")]
-    pub async fn commit_to_async<B>(
-        &mut self,
-        persist_backend: &mut B,
-    ) -> Result<bool, B::WriteError>
-    where
-        B: bdk_chain::persist::PersistBackendAsync<ChangeSet> + Send + Sync,
-    {
-        let committed =
-            bdk_chain::persist::StageExtAsync::commit_to(&mut self.stage, persist_backend).await?;
-        Ok(committed.is_some())
+    /// Get a reference of the staged [`ChangeSet`] that are yet to be committed (if any).
+    pub fn staged(&self) -> Option<&ChangeSet> {
```

Comment:
I thought the previous implementation of this was fine too.

Reply:
You mean without the option?

Reply:
I think I was worried that people will try persist using this and then call `take_staged` only after saving to db succeeded.